### PR TITLE
Tests Sessions for auth

### PR DESCRIPTION
**Pull request by cjavdev**
### TL;DR
Added RSpec tests for the Game model's word-guessing functionality.

### What changed?
- Created new test file `spec/game_spec.rb` with comprehensive test coverage for the Game model
- Added validation tests to ensure games require a word
- Implemented test cases for the `check_guess` method covering:
  - Invalid input handling (nil values, incorrect length)
  - Exact word matches
  - Partial letter matches
  - Handling of duplicate letters
- Adjusted terminal window configuration in demo.yml for better display

### How to test?
1. Run the test suite: `rspec spec/game_spec.rb`
2. Verify all test cases pass
3. Try adding new test cases for edge cases (e.g., invalid characters)

### Why make this change?
To ensure the Game model's word-guessing logic works correctly and maintains expected behavior through future changes. The tests cover core functionality like input validation, exact matches, partial matches, and special cases like duplicate letters.

### CODE REVIEW

**Comment by cjavdev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/cjavorg/scratch/38?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#38** <a href="https://app.graphite.dev/github/pr/cjavorg/scratch/38?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/cjavorg/scratch/38?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#23** <a href="https://app.graphite.dev/github/pr/cjavorg/scratch/23?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#37](https://github.com/cjavorg/scratch/pull/37) <a href="https://app.graphite.dev/github/pr/cjavorg/scratch/37?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#39](https://github.com/cjavorg/scratch/pull/39) <a href="https://app.graphite.dev/github/pr/cjavorg/scratch/39?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#22** <a href="https://app.graphite.dev/github/pr/cjavorg/scratch/22?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>merge-me</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/cjavorg?ref=merge-queue-instructions-in-stack-comment&prId=38).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: spec/game_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      end
+
+      it "raises error when guess length is not 5" do
+        expect { game.check_guess("ABC") }.to raise_error(ArgumentError, "Guess must be 5 letters")
+      end
+
+      it "handles partial matches" do
+        correct_result = [:partial, :partial, :incorrect, :correct, :partial] # Should be dynamically computed
+        expect(game.check_guess("PAPEL")).to eq(correct_result)
+      end
```
[the remark this answers is not in the document]
Comment (cjavdev) [reply spec/game_spec.rb:29]: Can you move them for me?
